SpatialGDK/Source/SpatialGDKServices/Private/SpatialCommandUtils.cpp

```diff
@@ -191,3 +191,55 @@ bool SpatialCommandUtils::GenerateDevAuthToken(bool bIsRunningInChina, FString&
 	OutTokenSecret = TokenSecret;
 	return true;
 }
+
+FProcHandle SpatialCommandUtils::StartLocalReceptionistProxyServer(bool bIsRunningInChina, const FString& CloudDeploymentName, FString &OutResult, int32 &OutExitCode)
+{
+	FString Command = TEXT("cloud connect external ");
+
+	Command += CloudDeploymentName;
+
+	if (bIsRunningInChina)
+	{
+		Command += SpatialGDKServicesConstants::ChinaEnvironmentArgument;
+	}
+
+	FProcHandle ProcHandle;
+
+	void* ReadPipe = nullptr;
+	void* WritePipe = nullptr;
+	ensure(FPlatformProcess::CreatePipe(ReadPipe, WritePipe));
+
+	ProcHandle = FPlatformProcess::CreateProc(*SpatialGDKServicesConstants::SpatialExe, *Command, false, true, true, nullptr, 1 /*PriorityModifer*/, *SpatialGDKServicesConstants::SpatialOSDirectory, WritePipe);
+
+	bool bProcessSucessed = false;
+	bool bProcessFinished = false;
+	if (ProcHandle.IsValid())
+	{
+		while(!bProcessFinished && !bProcessSucessed)
```

Comment:
is it possible for the proxy to never get into a good state? maybe by specifying an invalid listening address?

Basically: Do we need a time out for this?

Reply:
If the proxy does not run and an error is returned the process will be finished - so the bProcessFinished will return true so that is the error case scenario.